tripy/tripy/function_registry.py

```diff
@@ -385,6 +416,10 @@ def wrapper(*args, **kwargs):
 
                     self[key] = wrapper
                 else:
+                    assert not hasattr(
```

Comment:
Could you also add a test where we bypass dispatch for a function and then register an overload?

Reply:
Added.